siatest/renter/renter_test.go

```diff
@@ -311,6 +311,45 @@ func testRenterStreamingCache(t *testing.T, tg *siatest.TestGroup) {
 	// Grab the first of the group's renters
 	r := tg.Renters()[0]
 
+	// Testing setting downloadCacheSize for streaming
+	// Test setting it to larger than the defaultCacheSize
+	if err := r.RenterSetStreamCacheSizePost(4); err != nil {
```

Comment:
This test is pretty good but it also needs to check if increasing/decreasing the cache size actually increases/decreases the cache. So it would be good to set it to 2, upload a 2 chunk file and see if it is as fast as expected, then set it to 1 and check if the 2 chunk file takes longer to download now since setting the cache size to a smaller value sould prune the cache to that size right away. Finally set it to 3 and make sure you can download a 3 chunk file.

Reply:
As discussed, adding an additional unit test to verify cache is pruned when `cacheSize` is set to a smaller value, and making sure the cache is pruned from within the `SetStreamingCacheSize` method